### PR TITLE
feat(artifacts): add git https insecure option. Fixes #10762

### LINKED ISSUE #10762: Support disabling TLS verify for Git artifacts retrieval
# Summary

Support disabling TLS verify for Git artifacts retrieval.

# Use Cases

When working with non-trusted Git repositories, this option can be helpful. 

**Message from the maintainers**:

Love this enhancement proposal? Give it a 👍. We prioritise the proposals with the most 👍.

### PR DESCRIPTION
**Pull request by aslafy-z**
Fixes https://github.com/argoproj/argo-workflows/issues/10762

Please do not open a pull request until you have checked ALL of these:

* [x] Create the PR as draft .
* [ ] Run `make pre-commit -B` to fix codegen and lint problems.
* [x] Sign-off your commits (otherwise the DCO check will fail).
* [x] Use [a conventional commit message](https://www.conventionalcommits.org/en/v1.0.0/) (otherwise the commit message check will fail).
* [x] "Fixes #" is in both the PR title (for release notes) and this description (to automatically link and close the issue).
* [ ] Add unit or e2e tests. Say how you tested your changes. If you changed the UI, attach screenshots.
* [ ] Github checks are green.
* [ ] Once required tests have passed, mark your PR "Ready for review".

If changes were requested, and you've made them, dismiss the review to get it reviewed again.

### CODE REVIEW

File: pkg/apis/workflow/v1alpha1/workflow_types.go [hunk 1]
```diff
@@ -2390,6 +2390,9 @@ type GitArtifact struct {
 
 	// InsecureIgnoreHostKey disables SSH strict host key checking during git clone
 	InsecureIgnoreHostKey bool `json:"insecureIgnoreHostKey,omitempty" protobuf:"varint,8,opt,name=insecureIgnoreHostKey"`
+ 
+	// InsecureSkipTLS indicates if server's certificate will not be checked for validity. This makes the HTTPS connections insecure.
+	InsecureIgnoreHostKey bool `json:"insecureIgnoreHostKey,omitempty" protobuf:"varint,8,opt,name=insecureIgnoreHostKey"`
```
Comment (aslafy-z) [inline pkg/apis/workflow/v1alpha1/workflow_types.go:2395]:
```suggestion
	InsecureSkipTLS bool `json:"insecureSkipTLS,omitempty" protobuf:"varint,8,opt,name=insecureSkipTLS"`
```

**Comment by aslafy-z** [comment]
I'd like to work on this PR using a GitHub Codespace but the image build is failing with the following: [creation.log](https://github.com/argoproj/argo-workflows/files/11124978/creation.log). It looks like k3d is unable to pull images from ghcr.io. Is this something you seen already? Thank you

**Comment by tooptoop4** [comment]
where @aslafy-z

**Comment by aslafy-z** [comment]
@agilgur5 can you please re-open this PR? Thank you

**Pull request closed without merging**